dataplane/pkg/common/egress_interface.go

```diff
@@ -60,32 +60,27 @@ func findDefaultGateway4() (string, net.IP, error) {
 	}
 	defer f.Close()
 
-	reader := bufio.NewReader(f)
-	return parseProcFile(reader)
+	scanner := bufio.NewScanner(f)
+	return parseProcFile(scanner)
 }
 
-func parseProcFile(reader *bufio.Reader) (string, net.IP, error) {
-	for {
-		line, err := reader.ReadString('\n')
-		if err != nil {
-			if err != io.EOF {
-				break
-			}
+func parseProcFile(scanner *bufio.Scanner) (string, net.IP, error) {
+	for scanner.Scan() {
+		if err := scanner.Err(); err != nil {
 			logrus.Errorf("Failed to read routes files: %v", err)
 			break
 		}
-		if line == "" {
-			break
-		}
-		line = strings.TrimSpace(line)
-		parts := strings.Split(line, "\t")
-
-		if strings.TrimSpace(parts[1]) == "00000000" {
-			outgoingInterface := strings.TrimSpace(parts[0])
-			defaultGateway := strings.TrimSpace(parts[2])
-			ip := parseGatewayIP(defaultGateway)
-			logrus.Printf("Found default gateway %v outgoing: %v", ip.String(), outgoingInterface)
-			return outgoingInterface, ip, nil
+		line := strings.TrimSpace(scanner.Text())
+		if len(line) > 0 {
```

Comment:
Could you please invert that condition to avoid nesting of if-statements:
```go
if len(line) == 0 {
    continue;
}

parts := strings.Split(line, "\t")
...
```